src/components/designer/taro/actions/SendAssetModal.spec.tsx

```diff
@@ -0,0 +1,279 @@
+import React from 'react';
+import { fireEvent } from '@testing-library/dom';
+import { waitFor } from '@testing-library/react';
+import { Status } from 'shared/types';
+import { BitcoindLibrary } from 'types';
+import { initChartFromNetwork } from 'utils/chart';
+import {
+  defaultTaroAddress,
+  defaultTaroAsset,
+  getNetwork,
+  injections,
+  lightningServiceMock,
+  renderWithProviders,
+  taroServiceMock,
+} from 'utils/tests';
+import SendAssetModal from './SendAssetModal';
+
+const bitcoindServiceMock = injections.bitcoindService as jest.Mocked<BitcoindLibrary>;
+
+describe('SendAssetModal', () => {
+  let unmount: () => void;
+
+  const renderComponent = async () => {
+    const network = getNetwork(1, 'test network', Status.Started, 2);
+
+    const initialState = {
+      taro: {
+        nodes: {
+          'alice-taro': {
+            assets: [
+              defaultTaroAsset({ id: 'test1234', name: 'testAsset', amount: '1000' }),
+            ],
+            balances: [],
+          },
+        },
+      },
+      network: {
+        networks: [network],
+      },
+      designer: {
+        activeId: network.id,
+        allCharts: {
+          [network.id]: initChartFromNetwork(network),
+        },
+      },
+      modals: {
+        sendAsset: {
+          visible: true,
+          nodeName: 'alice-taro',
+        },
+      },
+    };
+    const cmp = <SendAssetModal network={network} />;
+
+    const result = renderWithProviders(cmp, { initialState, wrapForm: true });
+    unmount = result.unmount;
+    return {
+      ...result,
+      network,
+    };
+  };
+
+  afterEach(() => unmount());
+
+  describe('When the modal displays', () => {
+    it('should render labels', async () => {
+      const { getByLabelText } = await renderComponent();
+      expect(getByLabelText('Send Asset from alice-taro')).toBeInTheDocument();
+    });
+
+    it('should hide when cancel is clicked', async () => {
+      const { getByText, queryByText, store } = await renderComponent();
+      const btn = getByText('Cancel');
+      expect(btn).toBeInTheDocument();
+      expect(btn.parentElement).toBeInstanceOf(HTMLButtonElement);
+      fireEvent.click(btn);
+      await waitFor(() => {
+        expect(store.getState().modals.sendAsset.visible).toBe(false);
+        expect(queryByText('Cancel')).not.toBeInTheDocument();
+      });
+    });
+  });
+  describe('When lnd balance is low', () => {
+    beforeEach(() => {
+      lightningServiceMock.getBalances.mockResolvedValue({
+        confirmed: '100',
+        unconfirmed: '200',
+        total: '300',
+      });
+      taroServiceMock.decodeAddress.mockResolvedValue(
+        defaultTaroAddress({ id: 'test1234', type: 'NORMAL', amount: '100' }),
+      );
+    });
+    it('an alert should display', async () => {
```

Comment:
Tests should always start with the word "should".

Reply:
done